sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -237,6 +229,9 @@ case class Divide(left: Expression, right: Expression)
     }
   }
 
+  // Used by doGenCode
+  protected def divide(eval1: ExprCode, eval2: ExprCode, javaType: String): String
```

Comment:
you can see from [this test](https://github.com/apache/spark/pull/14036/files#diff-f3fc3b985a8877ee493f1a12ddfb98d6L143) that the previous `Divide` expression can support all integral types and fraction types.

BTW looks there is no special handling for integral type in your PR, for fraction type:
```
if (dataType.isInstanceOf[DecimalType]) {
  s"${eval1.value}.$decimalMethod(${eval2.value})"
} else {
  s"($javaType)(${eval1.value} $symbol ${eval2.value})"
}
```
while the `symbol` is `/`, so we generate `($javaType)(${eval1.value} / ${eval2.value})` for double and float.

for integral type:
```
($javaType)(${eval1.value} $decimalMethod (${eval2.value}))
```
while the `decimalMethod` is `/`, so it's still generates `($javaType)(${eval1.value} / ${eval2.value})`